der/src/asn1/general_string.rs

```diff
@@ -0,0 +1,30 @@
+use crate::{BytesRef, DecodeValue, EncodeValue, FixedTag, Header, Length, Reader, Tag, Writer};
+
+/// This is currently `OctetStringRef` internally, as `GeneralString` is not fully implemented yet
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct GeneralStringRef<'a> {
+    /// Raw contents, unchecked
+    #[doc(hidden)]
+    pub __contents: &'a [u8],
```

Comment:
The other ASN.1 string types use `BytesRef` for this, which has a precomputed `Length`, FWIW